src/common/locations.ts

```diff
@@ -186,3 +196,48 @@ export function getColleges(fips: string): CollegeData[] {
 export function getStateCode(stateName: string) {
   return REVERSED_STATES[stateName];
 }
+
+const ALL_LOCATIONS = getLocationNames();
+const [COUNTIES, STATES] = partition(ALL_LOCATIONS, isCounty);
+
+export function getStateByUrlName(stateUrlName: string): Location | undefined {
+  return STATES.find(
+    state =>
+      toLower(state.state_url_name) === toLower(stateUrlName) ||
+      toLower(state.state_code) === toLower(stateUrlName),
+  );
+}
+
+export function getCountyByUrlName(
+  countyUrlName: string,
+): Location | undefined {
+  return COUNTIES.find(
+    county => toLower(county.county_url_name) === toLower(countyUrlName),
+  );
+}
+
+export function getLocationByUrlParams(
+  stateUrlName: string,
+  countyUrlName?: string,
+): Location | undefined {
+  const location = countyUrlName
+    ? getCountyByUrlName(countyUrlName)
+    : getStateByUrlName(stateUrlName);
```

Comment:
Per comment above, I think we need to combine these, not use them separately since county URL names are not unique by themselves.